app/src/main/java/io/homeassistant/companion/android/util/DisabledLocationHandler.kt

```diff
@@ -0,0 +1,60 @@
+package io.homeassistant.companion.android.util
+
+import android.Manifest
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.location.LocationManager
+import android.os.Build
+import android.provider.Settings
+import androidx.appcompat.app.AlertDialog
+import io.homeassistant.companion.android.R
+
+object DisabledLocationHandler {
+    fun isLocationEnabled(context: Context): Boolean {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            // This is new method provided in API 28
+            val lm: LocationManager =
+                context.getSystemService(Context.LOCATION_SERVICE) as LocationManager
+            lm.isLocationEnabled
```

Comment:
So it was to my understanding that certain versions of Android allow the user to keep location enabled however the user can optionally turn off GPS.  I thought what we needed to do here was to also check:

`lm.isProviderEnabled(LocationManager.GPS_PROVIDER)`

Which tells us if GPS itself was disabled.  According to the users in https://github.com/home-assistant/android/issues/924#issuecomment-700430278 they are able to leave location on but GPS off.  Did you find similar results too?

Judging by the API it seems both calls can return different results unless I am misunderstanding this?

https://developer.android.com/reference/android/location/LocationManager#isProviderEnabled(java.lang.String)
https://developer.android.com/reference/android/location/LocationManager#isLocationEnabled()

Reply:
Yes is i think you are right. With Android P onwards it is enough to call `LocationManger.isLocationEnabled`. For everthing else we should call `LocationManger.isProviderEnabled(LocationManager.NETWORK_PROVIDER)` or `LocationManger.isProviderEnabled(LocationManager.GPS_PROVIDER)` depending on which permission is needed (COARSE or FINE).

I've updated the code to this end.